library/swift/RequestBuilder.swift

```diff
@@ -0,0 +1,106 @@
+import Foundation
+
+/// Builder used for constructing instances of `Request` types.
+@objcMembers
+public final class RequestBuilder: NSObject {
+  /// URL for the request.
+  public private(set) var url: URL
+  /// Method for the request.
+  public private(set) var method: RequestMethod
+  /// Headers to send with the request.
+  /// Multiple values for a given name are valid, and will be sent as comma-separated values.
+  public private(set) var headers: [String: [String]] = [:]
+  /// Trailers to send with the request.
+  /// Multiple values for a given name are valid, and will be sent as comma-separated values.
+  public private(set) var trailers: [String: [String]] = [:]
+  // Serialized data to send as the body of the request.
+  public private(set) var body: Data?
+  // Retry policy to use for this request.
+  public private(set) var retryPolicy: RetryPolicy?
+
+  // MARK: - Initializers
+
+  /// Internal initializer used for converting a request back to a builder.
+  init(request: Request) {
+    self.url = request.url
```

Comment:
https://github.com/lyft/envoy-mobile/pull/240/files#r302691380

(cross-posting comments from your two PRs since we'll ultimately want consistency)

Reply:
https://github.com/lyft/envoy-mobile/pull/240/files#r302696592 😁